core-bundle/src/Resources/contao/models/PageModel.php

```diff
@@ -1349,7 +1350,22 @@ public function getFrontendUrl($strParams=null, $strForceLang=null)
 		}
 
 		$objRouter = System::getContainer()->get('router');
-		$strUrl = $objRouter->generate(RouteObjectInterface::OBJECT_BASED_ROUTE_NAME, array(RouteObjectInterface::CONTENT_OBJECT => $this, 'parameters' => $strParams));
+
+		try
+		{
+			$strUrl = $objRouter->generate(RouteObjectInterface::OBJECT_BASED_ROUTE_NAME, array(RouteObjectInterface::CONTENT_OBJECT => $this, 'parameters' => $strParams));
+		}
+		catch (RouteNotFoundException $e)
+		{
+			$pageRegistry = System::getContainer()->get('contao.routing.page_registry');
+
+			if (!$pageRegistry->isRoutable($this))
+			{
+				throw new ResourceNotFoundException(sprintf('Page ID %s is not routable', $this->id));
```

Comment:
When rethrowing, the inner exception should IMHO always get passed on.

```suggestion
				throw new ResourceNotFoundException(sprintf('Page ID %s is not routable', $this->id), 0, $e);
```